src/Sarif.UnitTests/SarifVersionOneToCurrentTests.cs

```diff
@@ -317,60 +334,65 @@ public void SarifTransformerTests_ToCurrent_OneRunWitRules()
-            rule.Properties["some_key"].SerializedValue.Should().Be("\"FoxForceFive\"");
-
-            rule = run.Resources.Rules["C2002"];
-
-            rule.Id.Should().Be("C2002");
-            rule.Name.Should().BeNull();
-            rule.ShortDescription.Should().BeNull();
-            rule.FullDescription.Should().NotBeNull();
-            rule.FullDescription.Text.Should().Be("Catfish season continuous hen lamb include dose copy grant.");
-            rule.Configuration.Should().NotBeNull();
-            rule.Configuration.Enabled.Should().Be(true);
-            rule.Configuration.DefaultLevel.Should().Be(RuleConfigurationDefaultLevel.Error);
-            rule.HelpLocation.Should().NotBeNull();
-            rule.HelpLocation.Uri.ToString().Should().Be("http://www.domain.com/rules/c2002.html");
-            rule.MessageStrings.Should().BeNull();
-            rule.Properties.Should().BeNull();
-
-            rule = run.Resources.Rules["C2003"];
-
-            rule.Id.Should().Be("C2003");
-            rule.Name.Should().NotBeNull();
-            rule.Name.Text.Should().Be("Rule C2003");
-            rule.ShortDescription.Should().NotBeNull();
-            rule.ShortDescription.Text.Should().Be("Rules were meant to be broken.");
-            rule.FullDescription.Should().NotBeNull();
-            rule.FullDescription.Text.Should().Be("Rent internal rebellion competence biography photograph.");
-            rule.Configuration.Should().NotBeNull();
-            rule.Configuration.Enabled.Should().Be(false);
-            rule.Configuration.DefaultLevel.Should().Be(RuleConfigurationDefaultLevel.Note);
-            rule.HelpLocation.Should().BeNull();
-            rule.MessageStrings.Should().BeNull();
-            rule.Properties.Should().BeNull();
+            string v2LogText = JsonConvert.SerializeObject(v2Log, SarifTransformerUtilities.JsonSettingsV2);
+            string v2LogExpectedText =
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-2.0.0"",
+  ""version"": ""2.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""results"": [],
+      ""resources"": {
+        ""rules"": {
+          ""C2001"": {
+            ""id"": ""C2001"",
+            ""shortDescription"": {
+              ""text"": ""A variable was used without being initialized.""
+            },
+            ""messageStrings"": {
+              ""default"": ""Variable \""{0}\"" was used without being initialized.""
+            },
+            ""properties"": {
+              ""some_key"": ""FoxForceFive""
+            }
+          },
+          ""C2002"": {
+            ""id"": ""C2002"",
+            ""fullDescription"": {
+              ""text"": ""Catfish season continuous hen lamb include dose copy grant.""
+            },
+            ""configuration"": {
+              ""enabled"": true,
+              ""defaultLevel"": 2
```

Comment:
The value should be the string `"error"`. Are we missing an `EnumHint` in CodeGenHints.json?

Reply:
And see how valuable it was to change the test to this form!

Reply:
Fixed. Missing ContractResolver case for RuleConfigurationDefaultLevel